lib/src/indexer.dart

```diff
@@ -15,35 +18,55 @@ Future<Index> indexPackage(
 ) async {
   final dirPath = p.normalize(p.absolute(root));
 
+  final rootHasPubspecFile = await File(
+    p.join(dirPath, 'pubspec.yaml'),
+  ).exists();
+  if (!rootHasPubspecFile) {
+    stderr.writeln(
+      'Provided path does not contain a pubspec.yaml file. '
+      'Unable to index',
+    );
+    exit(1);
+  }
+
   final metadata = Metadata(
-      projectRoot: 'file:/' + dirPath,
-      textDocumentEncoding: TextEncoding.UTF8,
-      toolInfo: ToolInfo(
-        name: 'scip-dart',
-        version: '0.0.1',
-        arguments: [],
-      ));
+    projectRoot: 'file:/' + dirPath,
+    textDocumentEncoding: TextEncoding.UTF8,
+    toolInfo: ToolInfo(
+      name: 'scip-dart',
+      version: '0.0.1',
```

Comment:
Unrelated but should this match the version of this library? Do we need to move this to a constant somewhere and configure the release event to bump it as well?

Reply:
Yea, ideally we do

I didn't love the idea of relying on a release event (if for whatever reason scip-dart changed owners outside of workiva one day) and wanted the solution to be baked into the project

I have an issue tracking this [here](https://github.com/Workiva/scip-dart/issues/43) and a spike pr for one of the solutions here: https://github.com/Workiva/scip-dart/pull/45. Just haven't pushed on wrapping that up yet